skill-tree /src/main/java/com/RDS/skilltree/Endorsement/EndorsementServiceImpl.java

```diff
@@ -0,0 +1,24 @@
+package com.RDS.skilltree.Endorsement;
+import org.springframework.stereotype.Service;
+
+import java.util.*;
+
+@Service
+public class EndorsementServiceImpl implements EndorsementService {
+    private final EndorsementRepository endorsementRepository;
+
+    public EndorsementServiceImpl(EndorsementRepository endorsementRepository) {
+        this.endorsementRepository = endorsementRepository;
+    }
+
+    @Override
+    public EndorsementDTO getEndorsementById(UUID id) throws IllegalStateException {
+        Optional<EndorsementModel> endorsementModel = endorsementRepository.findById(id);
+        if (endorsementModel.isPresent()) {
+            return EndorsementDTO.toDto(endorsementModel.get());
+        } else {
+            String errorMessage = "No endorsement with the id " + id + " found";
+            throw new IllegalStateException(errorMessage);
+        }
```

Comment:
When working with `Optional` try to use the `.orElseThrow` method to handle cases of
```
 if present
   then do something
 else
   throw exception
```
```suggestion
        return EndorsementDTO.toDto(endorsementModel.orElseThrow(() -> new IllegalStateException("No endorsement with the id " + id + " found")));
```

Reply:
Thanks for suggestion.Implemented as suggested.